shumai/io/network.ts

```diff
@@ -19,27 +19,35 @@ export function encode(tensor: sm.Tensor): ArrayBuffer {
   return buf.buffer
 }
 
-export function decode(obj: Response | ArrayBuffer): sm.Tensor {
-  function impl(buf) {
-    const shape_len = new Int32Array(buf, 0, 2)[0]
-    const shape = new BigInt64Array(buf, 8 /* 8 byte offset mandated alignement */, shape_len)
-    const t = sm.tensor(new Float32Array(buf, 8 + 8 * shape_len))
-    return t.reshape(shape)
-  }
+export function decodeBuffer(buf: ArrayBuffer) {
+  const shape_len = new Int32Array(buf, 0, 2)[0]
+  const shape = new BigInt64Array(buf, 8 /* 8 byte offset mandated alignement */, shape_len)
+  const t = sm.tensor(new Float32Array(buf, 8 + 8 * shape_len))
+  return t.reshape(shape as unknown as number[])
```

Comment:
shape is always a BigInt64Array in this case, what does `as number[]` do?

Reply:
Error: `Argument of type 'BigInt64Array' is not assignable to parameter of type 'number[]'`

This resolves the error by asserting that the type of the param is of `number[]`; it probably is better to update the types for `reshape` to allow passing in `BigInt64Array | number[]`.